src/AxisSets.jl

```diff
@@ -1,5 +1,218 @@
 module AxisSets
 
-# Write your package code here.
+using AxisKeys
+using Impute
+using IterTools
+using NamedDims
+using OrderedCollections
+using ReadOnlyArrays
+using Tables
+
+using Impute:
+    Filter,
+    Imputor,
+    Validator
+
+# Short hand type for complicated union of nested Keyed or NamedDims arrays
+XArray{L, T, N} = Union{NamedDimsArray{L,T,N,<:KeyedArray}, KeyedArray{T,N,<:NamedDimsArray}}
+
+struct Dataset{T<:XArray}
+    dims::OrderedSet{Symbol}
+    data::LittleDict{Symbol, T}
+end
+
+function Dataset(dims::Symbol...; kwargs...)
+    # If we didn't get any kwargs then just create an empty Dataset
+    isempty(kwargs) && return Dataset(OrderedSet(dims), LittleDict{Symbol, XArray}())
+
+    # Checking the eltype doesn't always work if the `NamedDimsArray` covers different dimensions
+    if all(v -> isa(v, XArray), values(kwargs))
+        # If we were passed a valid XArray then just construct our LittleDict
+        data = LittleDict{Symbol, XArray}(kwargs...)
+    else
+        # Otherwise we need to assume that we're operating over a collection of tables
+        data = LittleDict{Symbol, XArray}()
+
+        # Might be a simpler way to do this if we have a `merge` function or something
+        for (name, table) in kwargs
+            @show typeof(table)
+            # Reuse the table constructor and extra the data
```

Comment:
this is because `table` might be a NamedTuple?

Reply:
No, just that if the kwargs aren't KeyedArrays then we need to fallback to the more general table constructor. Since we want that constructor anyways, seemed like we should just reuse it here.